packages/keyda-demo/src/lib/FormToSubmit/index.js

```diff
@@ -2,39 +2,58 @@ import React, { useCallback } from 'react';
 import _ from 'lodash';
 import axios from 'axios';
 
-import { useKeyStateState } from '../Context';
+import { useKeyStateState, useKeyStateDispatch } from '../Context';
 
 const FormToSubmit = (props) => {
-  const definedFormTypes = ['REGISTER', 'LOGIN', 'PASSWORD_CHANGE'];
-
+  const definedFormTypes = ['REGISTER', 'LOGIN', 'CHANGE_PW'];
   const { onSubmit, formType, children, ...rest } = props;
 
   if (!formType) {
     throw new Error(
-      'You should pass the "formType" prop explicitly to define a mechanism of KeyDa form.\nYou can choose one of "REGISTER", "LOGIN", "PASSWORD_CHANGE" matched with your usage.'
+      'You should pass the "formType" prop explicitly to define a mechanism of KeyDa form.\nYou can choose one of "REGISTER", "LOGIN", "CHANGE_PW" matched with your usage.'
     );
   } else if (!_.includes(definedFormTypes, formType)) {
     throw new Error(
-      'You passed wrong "formType". You can choose one of "REGISTER", "LOGIN", "PASSWORD_CHANGE" matched with your usage.'
+      'You passed wrong "formType". You can choose one of "REGISTER", "LOGIN", "CHANGE_PW" matched with your usage.'
     );
   }
 
+  const REQUEST_URL = 'http://localhost:5000/api/users/'; // temporary request url : localhost of keyda-server
+  const suffix = formType.toLowerCase();
+
   const keyState = useKeyStateState();
+  const keyDispatch = useKeyStateDispatch();
 
   const handleSubmit = useCallback(
-    async (e) => {
+    (e) => {
       e.preventDefault();
-      const formData = new FormData();
-      console.log(formData);
-      onSubmit();
-      const post = await axios.post(
-        'http://localhost:5000/api/users/register', // temporary request url : localhost of keyda-server
-        keyState
-      );
-      console.log(post);
-      console.log(keyState);
+      setTimeout(() => {
+        const formData = new FormData();
+        console.log(formData);
+        formData.append('keyTimeList', keyState.keyTimeList);
+        formData.append('userId', keyState.userId);
+
+        axios.post(REQUEST_URL + suffix, formData).then((response) => {
+          console.log(response);
+        });
+
+        keyDispatch({
+          type: 'REGISTER',
+        });
+
+        console.log(keyState);
+        if (keyState.trainCount + 1 === 5) {
+          keyDispatch({
+            type: 'SUBMIT',
+          });
+          onSubmit();
+        }
+      }, 500);
```

Comment:
When we use `async/await` in the event handler function, the event object is not re-usable anymore.
So I changed the logic for `onSubmit` callback function to using `setTimeout` instead of storing response.
And in fact, we didn't need to store the `post` response.
https://ko.reactjs.org/docs/events.html#event-pooling